### ISSUE TITLE
RightCurly: Google style config does not detect misplaced '}' for annotation declaration and enum

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#11629]
I have read check documentation: https://checkstyle.org/config_blocks.html#RightCurly
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

detected at https://github.com/checkstyle/checkstyle/pull/11585#discussion_r866844938
this input file is part of our regression code base:
```bash
$ cat Test.java 
class Test {
  public @interface TestAnnotation {}

  public @interface TestAnnotation1 { String someValue(); }

  public @interface TestAnnotation2 {
    String someValue(); }  // line 7, it should be warn !!!

  public @interface TestAnnotation3 {
    String someValue();
  } //ok

  public @interface TestAnnotation4 { String someValue();
  } //ok

  enum TestEnum2 {
    SOME_VALUE; } // line 17, it should be warn !!!
}

$ java -classpath checkstyle-10.2-all.jar com.puppycrawl.tools.checkstyle.Main \
        -c /google_checks.xml Test.java 
Starting audit...
Audit done.
```

**Describe what you expect in detail.**

there should be violation on line 7 and 17.
from google style https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.1.2-blocks-k-r-style
>Line break before the closing brace

so we have defect.

removal of following suppression module from google config make violations appear on lines:
```
    <module name="SuppressionXpathSingleFilter">
      <!-- suppresion is required till https://github.com/checkstyle/checkstyle/issues/7541 -->
      <property name="id" value="RightCurlyAlone"/>
      <property name="query" value="//RCURLY[parent::SLIST[count(./*)=1]
                                     or preceding-sibling::*[last()][self::LCURLY]]"/>
    </module>

```

**Comment by Kevin222004** [comment]
i am on it

**Comment by Kevin222004** [comment]
@romani I have looked at the main code base but the code looks valid to me i don't understand what to fix and how.

**Comment by romani** [comment]
There is some problem in suppression. It was done to hide some defect, and looks like  suppress too much violations as this issue state. 
We need to find way to update xpath to not suppress undesirable violations or fix https://github.com/checkstyle/checkstyle/issues/7541 and remove suppression.

**Comment by rnveach** [comment]
@romani I still recommend https://github.com/checkstyle/checkstyle/issues/4086 as the way to move RIghtCurly further, instead of trying to make check after check imitations of it.

**Comment by Youssefalaa7** [comment]
I am on it

**Comment by romani** [comment]
not a google style issue any more after introduction of https://github.com/checkstyle/checkstyle/issues/7541

We need to help to run latest gogole style version on this Test.java to see if there is violation show up
